src/Workspaces/CSharp/Portable/CodeGeneration/ExpressionGenerator.cs

```diff
@@ -52,18 +50,19 @@ private static ExpressionSyntax GenerateNullLiteral()
             => SyntaxFactory.LiteralExpression(SyntaxKind.NullLiteralExpression);
 
         internal static ExpressionSyntax GenerateExpression(
-            ITypeSymbol type,
-            object value,
+            ITypeSymbol? type,
+            object? value,
```

Comment:
is there any NotNullWhenNotNull that says that type is not null if value is not null?